python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -235,7 +236,10 @@ def evaluate(self, *, cache: CSECache, timer: Timer | None) -> DataFrame:
             timer.store(start, end, type(self).__name__)
             return result
         else:
-            return self.do_evaluate(*self._non_child_args, *children)
+            args = (*self._non_child_args, *children)
+            return do_evaluate_with_tracing(self.do_evaluate, name=type(self).__name__)(
+                *args
+            )
```

Comment:
(If you go with my suggested change to `do_evaluate_with_tracing`)

```suggestion
            return do_evaluate_with_tracing(self)(
                *self._non_child_args, *children
            )
```